src/promql/src/planner.rs

```diff
@@ -1409,58 +1581,148 @@ impl PromPlanner {
+                        let right_table_ref = rhs_plan
+                            .schema()
+                            .field(0)
+                            .qualifier()
+                            .map(|x| x.to_owned_reference())
+                            .unwrap_or_else(|| OwnedTableReference::bare(""));
+                        let left_field_columns = vec![lhs_plan.schema().field(1).name().clone()];
+                        let right_field_columns = vec![rhs_plan.schema().field(1).name().clone()];
+                        let plan = self.join_time_series(
+                            left_table_ref,
+                            right_table_ref,
+                            left_field_columns.into_iter(),
+                            right_field_columns.into_iter(),
+                            lhs_plan,
+                            rhs_plan,
+                            is_comparison_op,
+                            should_return_bool,
+                            *op,
+                        )?;
+                        return Ok(Some(LiteralExpr::Runtime(plan)));
+                    }
+                    (lhs, rhs) => {
+                        let expr_builder = Self::prom_token_to_binary_expr_builder(*op)?;
+                        let expr = expr_builder(lhs.get_expr(), rhs.get_expr())?;
+                        let expr = if is_comparison_op && should_return_bool {
+                            DfExpr::Cast(Cast {
+                                expr: Box::new(expr),
+                                data_type: ArrowDataType::Float64,
+                            })
+                        } else {
+                            expr
+                        };
+                        let literal = match (lhs, rhs) {
+                            (LiteralExpr::Const(_), LiteralExpr::Const(_)) => {
+                                LiteralExpr::Const(expr)
+                            }
+                            (LiteralExpr::Const(_), LiteralExpr::Runtime(plan))
+                            | (LiteralExpr::Runtime(plan), LiteralExpr::Const(_)) => {
+                                let time_index = plan.expressions()[0].clone();
+                                self.ctx.field_columns =
+                                    vec![expr.display_name().context(DataFusionPlanningSnafu)?];
+                                LiteralExpr::Runtime(
+                                    LogicalPlanBuilder::from(plan)
+                                        .project(vec![time_index, expr])
+                                        .context(DataFusionPlanningSnafu)?
+                                        .build()
+                                        .context(DataFusionPlanningSnafu)?,
+                                )
+                            }
+                            (LiteralExpr::Runtime(_), LiteralExpr::Runtime(_)) => {
+                                unreachable!()
```

Comment:
I am confused about why it's unreachable. Could you explain it? It would be great to add some comments for it.

Reply:
Mainly to reduce some repetitive code，`LiteralExpr::Runtime(_), LiteralExpr::Runtime(_)` has been dispose by upper match, a comment has add on it